### PR TITLE
Move porous flow unit tests to porous_flow/unit

### LINKED ISSUE #14815: Move porous flow unit tests into module
## Reason
Unit tests should be located within their module (see #14807)

## Design
Move porous flow unit tests into porous flow module.

## Impact
Better developer experience: unit test would be where people would intuitively look for them.

### PR DESCRIPTION
**Pull request by cpgr**
Closes #14815

### CODE REVIEW

File: large_media [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 8e8b43ed3fe0649b5134313a2bdb3e36a3398075
+Subproject commit a93d9ac707babeda32ac3a01f710718f18a73a85
```
[the remark this answers is not in the document]
Comment (cpgr) [reply large_media]: I think I fixed this

**Comment by cpgr** [comment]
Hey @andrsd, I have a 'not a registered object' problem and can't work out a fix. Any ideas?

**Comment by andrsd** [comment]
The c-tor of `PorousFlowApp` does not call `registerAll`.

Apply this patch to fix the problem:
```
--- a/modules/porous_flow/src/base/PorousFlowApp.C
+++ b/modules/porous_flow/src/base/PorousFlowApp.C
@@ -31,7 +31,10 @@ validParams<PorousFlowApp>()
 
 registerKnownLabel("PorousFlowApp");
 
-PorousFlowApp::PorousFlowApp(const InputParameters & parameters) : MooseApp(parameters) {}
+PorousFlowApp::PorousFlowApp(const InputParameters & parameters) : MooseApp(parameters)
+{
+  PorousFlowApp::registerAll(_factory, _action_factory, _syntax);
+}
 
 PorousFlowApp::~PorousFlowApp() {}
```

**Comment by cpgr** [comment]
Oh wow, I would have never found that out! I hope it didn't take long, and thanks!

**Comment by andrsd** [comment]
> Oh wow, I would have never found that out! I hope it didn't take long, and thanks!

It took about 15 minutes...

**Comment by cpgr** [comment]
I'll have to shout you a beer next time I'm in America 🍻

**Review by andrsd — APPROVED** [review]

**Pull request merged**